x-pack/elastic-agent/pkg/agent/application/upgrade/step_download.go

```diff
@@ -18,9 +18,20 @@ import (
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/artifact/download/snapshot"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
+	"go.elastic.co/apm"
 )
 
-func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (string, error) {
+func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (_ string, err error) {
+	// TODO: span type?
+	span, ctx := apm.StartSpan(ctx, "Upgrader.downloadArtifact()", "custom")
+	defer func() {
+		span.End()
+		// TODO: do we want to capture the errors at this top-level, in
+		// the lower levels, or at each level?
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
+	}()
```

Comment:
this seemed like it would be interesting, but I seemingly haven't triggered a download yet.

Reply:
If you delete the bundled apm-server artifact from the image, it should download it when the apm integration is assigned